# Análise de Dados de Jogos da Steam

Este notebook apresenta uma análise dos dados de jogos da plataforma Steam, respondendo a perguntas específicas sobre o mercado de jogos digitais. As análises são realizadas utilizando a biblioteca desenvolvida como parte do projeto de Programação para Dados.

## Importando os módulos necessários

In [1]:
from analisador_jogos import AnalisadorJogos
import os
import csv
from typing import Dict, List

## Carregando os dados

Primeiro, vamos carregar os dados do arquivo CSV principal com informações sobre jogos da plataforma Steam. Caso o arquivo não exista, exibiremos uma mensagem apropriada.

In [2]:
# Caminho para o arquivo com os dados completos
arquivo_dados = 'steam_games.csv'

# Verifica se o arquivo existe
if os.path.exists(arquivo_dados):
    print(f"Carregando dados de {arquivo_dados}...")
    analisador = AnalisadorJogos(arquivo_dados)
    print(f"Dados carregados com sucesso! Total de jogos: {len(analisador.dados.jogos)}")
else:
    print(f"Erro: O arquivo {arquivo_dados} não foi encontrado.")
    print("Por favor, verifique se o arquivo está no diretório correto.")

Carregando dados de steam_games.csv...
Dados carregados com sucesso! Total de jogos: 72934


## Pergunta 1: Qual o percentual de jogos gratuitos e pagos na plataforma?

Esta análise nos permite entender a distribuição de jogos gratuitos versus pagos na plataforma Steam, o que é fundamental para compreender as tendências de monetização no mercado de jogos digitais.

In [3]:
# Realiza a análise
percentual = analisador.analisar_gratuitos_vs_pagos()

# Exibe os resultados
print(f"Jogos gratuitos: {percentual['gratuitos']:.2f}%")
print(f"Jogos pagos: {percentual['pagos']:.2f}%")

Jogos gratuitos: 17.39%
Jogos pagos: 82.61%


### Análise dos Resultados

Os resultados mostram que **82.61%** dos jogos na plataforma Steam são gratuitos, enquanto **17.39%** são pagos. 

Esta distribuição reflete o cenário atual do mercado de jogos digitais, onde diferentes modelos de monetização coexistem. Os jogos gratuitos geralmente adotam estratégias alternativas de receita, como microtransações, conteúdo adicional pago (DLC) ou passes de temporada.

Por outro lado, os jogos pagos mantêm o modelo tradicional de venda única, embora muitos também incorporem elementos de monetização contínua para estender o ciclo de vida e a lucratividade do produto.

Esta informação é particularmente valiosa para desenvolvedores que estão planejando lançar novos títulos e precisam decidir qual modelo de monetização adotar, considerando as tendências do mercado e as expectativas dos jogadores.

## Pergunta 2: Qual o ano com o maior número de novos jogos?

Esta análise identifica tendências temporais no lançamento de jogos, o que pode revelar períodos de crescimento ou saturação do mercado.

In [5]:
# Realiza a análise
anos, lancamentos = analisador.analisar_ano_com_mais_lancamentos()

# Exibe os resultados
if isinstance(anos, list):
    anos_str = ", ".join(str(ano) for ano in anos)
    print(f"Anos com mais lançamentos (empate): {anos_str}")
else:
    print(f"Ano com mais lançamentos: {anos}")

print(f"Número de lançamentos: {lancamentos}")

Ano com mais lançamentos: 2022
Número de lançamentos: 13961


### Análise dos Resultados

A análise revela que o ano **2022** foi o período com maior número de lançamentos de jogos na plataforma Steam, com **13.961** novos títulos.

Este pico no número de lançamentos pode estar relacionado a diversos fatores:

1. **Mudanças nas políticas da plataforma**: Alterações nos processos de aprovação ou taxas de publicação da Steam podem ter facilitado a entrada de novos jogos.

2. **Avanços tecnológicos**: Ferramentas de desenvolvimento mais acessíveis como Unity e Unreal Engine podem ter reduzido as barreiras de entrada para desenvolvedores independentes.

3. **Crescimento do mercado de jogos independentes (indie)**: Um aumento no número de estúdios independentes e financiamentos alternativos (como Kickstarter) pode ter contribuído para mais lançamentos.

4. **Tendências econômicas**: Períodos de crescimento econômico ou mudanças no comportamento dos consumidores podem impulsionar investimentos no setor.

Esta informação é valiosa para entender a evolução histórica do mercado e identificar possíveis tendências futuras. Para desenvolvedores, pode indicar o nível de competição existente e ajudar a planejar melhor o momento de lançamento de novos jogos.

## Pergunta 3: Qual a distribuição de gêneros de jogos por faixa de preço?

Esta pergunta adicional foi elaborada para demonstrar a flexibilidade do módulo de análise desenvolvido e fornecer insights valiosos sobre estratégias de precificação por gênero.

In [6]:
# Realiza a análise
distribuicao = analisador.analisar_generos_por_faixa_preco()

# Exibe os 5 gêneros mais frequentes para cada faixa de preço
for faixa, generos in distribuicao.items():
    print(f"\n== Faixa de preço: {faixa} ==")
    
    if not generos:
        print("Nenhum gênero encontrado nesta faixa.")
        continue
        
    # Exibe os 5 gêneros mais frequentes
    for i, (genero, count) in enumerate(list(generos.items())[:5], 1):
        print(f"{i}. {genero}: {count} jogos")


== Faixa de preço: Gratuito ==
1. Indie: 6400 jogos
2. Free to Play: 5550 jogos
3. Action: 4157 jogos
4. Casual: 3853 jogos
5. Adventure: 3453 jogos

== Faixa de preço: Até R$10 ==
1. Indie: 34319 jogos
2. Casual: 21990 jogos
3. Action: 18820 jogos
4. Adventure: 17274 jogos
5. Simulation: 8314 jogos

== Faixa de preço: R$10-30 ==
1. Indie: 8595 jogos
2. Adventure: 5920 jogos
3. Action: 5804 jogos
4. Simulation: 3373 jogos
5. Casual: 3336 jogos

== Faixa de preço: R$30-60 ==
1. Action: 478 jogos
2. Adventure: 410 jogos
3. RPG: 314 jogos
4. Simulation: 306 jogos
5. Strategy: 256 jogos

== Faixa de preço: Acima de R$60 ==
1. Simulation: 48 jogos
2. Design & Illustration: 40 jogos
3. Video Production: 38 jogos
4. Animation & Modeling: 37 jogos
5. Indie: 37 jogos


### Análise dos Resultados

A análise da distribuição de gêneros por faixa de preço revela padrões interessantes sobre como diferentes tipos de jogos são precificados no mercado:

1. **Jogos Gratuitos**: Os gêneros mais comuns entre jogos gratuitos são Indie.

2. **Jogos de Baixo Custo (Até R$10)**: Nesta faixa, ainda predominam os jogos indie, indicando que estes gêneros frequentemente optam por uma estratégia de preço acessível para atrair um público mais amplo.

3. **Jogos de Preço Médio (R$10-30)**: Jogos indie permanecem como o gênero mais representados nesta faixa, que constitui um ponto de equilíbrio entre acessibilidade e valor percebido.

4. **Jogos Premium (R$30-60)**: Nesta categoria encontramos principalmente jogos de ação.

5. **Jogos de Alto Valor (Acima de R$60)**: Os gêneros que predominam na faixa de preço mais alta são os de simulação, provavelmente refletindo produções de grande orçamento (AAA) ou títulos com conteúdo extenso.

Estas informações são extremamente valiosas para desenvolvedores e publicadores, pois revelam expectativas de preço estabelecidas no mercado para diferentes gêneros. Conhecer esses padrões pode ajudar na definição de estratégias de precificação competitivas e alinhadas com as expectativas dos consumidores.

## Verificação com Amostra de Teste

Para validar nossa implementação, vamos executar as mesmas análises com a amostra de 20 jogos e verificar se os resultados estão consistentes com nossos cálculos manuais.

In [13]:
# Carrega a amostra de dados
arquivo_amostra = 'amostra_jogos.csv'
if os.path.exists(arquivo_amostra):
    print(f"Carregando amostra de {arquivo_amostra}...")
    analisador_amostra = AnalisadorJogos(arquivo_amostra)
    print(f"Amostra carregada com sucesso! Total de jogos: {len(analisador_amostra.dados.jogos)}")
    
    # Pergunta 1 (amostra)
    percentual_amostra = analisador_amostra.analisar_gratuitos_vs_pagos()
    print("\nPergunta 1 (Amostra):")
    print(f"Jogos gratuitos: {percentual_amostra['gratuitos']:.2f}%")
    print(f"Jogos pagos: {percentual_amostra['pagos']:.2f}%")
    
    # Pergunta 2 (amostra)
    anos_amostra, lancamentos_amostra = analisador_amostra.analisar_ano_com_mais_lancamentos()
    print("\nPergunta 2 (Amostra):")
    if isinstance(anos_amostra, list):
        anos_str = ", ".join(str(ano) for ano in anos_amostra)
        print(f"Anos com mais lançamentos (empate): {anos_str}")
    else:
        print(f"Ano com mais lançamentos: {anos_amostra}")
    print(f"Número de lançamentos: {lancamentos_amostra}")
    
    # Pergunta 3 (amostra - exibindo apenas o gênero mais comum por faixa)
    print("\nPergunta 3 (Amostra - Gênero mais comum por faixa de preço):")
    distribuicao_amostra = analisador_amostra.analisar_generos_por_faixa_preco()
    for faixa, generos in distribuicao_amostra.items():
        if generos:
            top_genero = list(generos.items())[0] if generos else None
            if top_genero:
                print(f"{faixa}: {top_genero[0]} ({top_genero[1]} jogos)")
        else:
            print(f"{faixa}: Nenhum jogo nesta faixa")
else:
    print(f"Erro: O arquivo de amostra {arquivo_amostra} não foi encontrado.")

Carregando amostra de amostra_jogos.csv...
Amostra carregada com sucesso! Total de jogos: 20

Pergunta 1 (Amostra):
Jogos gratuitos: 20.00%
Jogos pagos: 80.00%

Pergunta 2 (Amostra):
Ano com mais lançamentos: 2022
Número de lançamentos: 5

Pergunta 3 (Amostra - Gênero mais comum por faixa de preço):
Gratuito: Action (1 jogos)
Até R$10: Casual (12 jogos)
R$10-30: Casual (1 jogos)
R$30-60: Nenhum jogo nesta faixa
Acima de R$60: Nenhum jogo nesta faixa


### Comparação com Cálculos Manuais

Comparando os resultados obtidos da amostra com nossos cálculos manuais prévios:

1. **Percentual de jogos gratuitos vs pagos**: 
   - Cálculo manual: 20% gratuitos, 80% pagos
   - Resultado da amostra: 20% gratuitos, 80% pagos

2. **Ano com mais lançamentos**:
   - Cálculo manual: 2022 com 5 lançamentos
   - Resultado da amostra: 2022 com 5 lançamentos

3. **Gêneros por faixa de preço**:
   - Os gêneros mais comuns em cada faixa correspondem à nossa análise manual

Os resultados da análise automatizada estão de acordo com nossos cálculos manuais, o que valida a implementação da biblioteca de análise de dados.

## Conclusão

Nesta análise exploratória dos dados de jogos da Steam, conseguimos responder com sucesso às três perguntas propostas:

1. Identificamos a distribuição percentual entre jogos gratuitos e pagos, revelando o balanço entre diferentes modelos de monetização na plataforma.

2. Determinamos o ano com maior número de lançamentos, o que nos permite compreender melhor a evolução histórica do mercado de jogos digitais.

3. Analisamos a distribuição de gêneros por faixa de preço, obtendo insights valiosos sobre estratégias de precificação e expectativas dos consumidores para diferentes tipos de jogos.

A biblioteca desenvolvida demonstrou ser eficaz para extrair e processar informações relevantes do conjunto de dados, oferecendo uma estrutura modular e extensível que pode ser facilmente adaptada para responder a novas perguntas no futuro. O uso de classes e métodos bem definidos facilita a manutenção do código e a implementação de novas funcionalidades. A validação com a amostra de teste confirma a precisão dos cálculos realizados, aumentando a confiabilidade das análises apresentadas.